### PR TITLE
[release/1.15.3] Add cloud stanza documentation

### PR DESCRIPTION
**Pull request by jjacobson93**
### Description

This PR adds documentation for the HCP cloud stanza.

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

**Review by hanshasselberg — COMMENTED** [review]
left two comments.

**Review by jmurret — COMMENTED** [review]
Should we clarify the title and description of this section that this is for connecting HCP Consul self-managed clusters (and possibly a sentence about it) vs "HCP"?

File: website/content/docs/agent/config/config-files.mdx [hunk 1]
```diff
@@ -1088,6 +1088,22 @@ Refer to the [formatting specification](https://golang.org/pkg/time/#ParseDurati
...
+
+- `cloud` This object specifies settings for connecting to HCP. This was added in Consul 1.14
+
+  - `client_id` The OAuth2 client ID for authentication with HCP
+
+  - `client_secret` The OAuth2 client secret for authentication with HCP
+
+  - `resource_id` The HCP resource identifier
+
+  - `hostname` (optional) The HCP hostname
```
[the remark this answers is not in the document]
Comment (jmurret) [reply website/content/docs/agent/config/config-files.mdx:1101]: I would vote for removing these as distracting.  I would wait until someone asks "why aren't these documented?"

**Review by hanshasselberg — APPROVED** [review]
LGTM

**Pull request merged**